frb/surveys/catalog_utils.py

```diff
@@ -388,7 +389,85 @@ def convert_mags_to_flux(photometry_table, fluxunits='mJy'):
         fluxtable[mag][uplimit] = fluxtable[mag][uplimit] 
 
     return fluxtable
+
+def remove_duplicates(tab:Table, idcol:str)->Table:
+    """
+    In an astropy table if there are duplicate
+    entries, remove the duplicates.
+
+    Args:
+        tab (Table): A table of entries.
+        idcol (str): A column name that has unique ids
+            for each table entry.
+    Returns:
+        unique_tab (Table): A table with only the unique ids.
+    """
+    assert isinstance(tab, Table), "Please provide an astropy table."
+    assert isinstance(idcol, str), "Please provide a valid column name."
+    assert idcol in tab.colnames, "{} not a column in the given table".format(idcol)
+    # Sort entries first.
+    tab.sort(idcol)
+    # Get the duplicates.
+    duplicate_ids = np.where(tab[1:][idcol]==tab[:-1][idcol])[0]+1
+    tab.remove_rows(duplicate_ids)
+    return tab
+    
+def xmatch_and_merge_cats(tab1:Table, tab2:Table, tol:units.Quantity=1*units.arcsec,
+                        table_names:tuple=('1','2'), **kwargs)->Table:
+    """
+    Given two source catalogs, cross-match and merge them. This function 
+    ensures there is a unique match between tables as opposed to the default join_skycoord
+    behavior which matches multiple objects on the right table to
+    a source on the left.
+    Args:
+        tab1, tab2 (Table): Photometry catalogs. Must contain columns named
+            ra and dec.
+        tol (Quantity[Angle], optional): Maximum separation for cross-matching.
+        table_names (tuple of str, optional): Names of the two tables for
+            naming unique columns in the merged table.
+        kwargs: Additional keyword arguments to be passed onto xmatch_catalogs
+    Returns:
+        merged_table (Table): Merged catalog.
+    """
+    if table_names is not None:
+        assert len(table_names)==2, "Invalid number of table names for two tables."
+        assert (type(table_names[0])==str)&(type(table_names[1])==str), "Table names should be strings."
     
+    assert np.all(np.isin(['ra','dec'],tab1.colnames)), "Table 1 doesn't have column 'ra' and/or 'dec'."
+    assert np.all(np.isin(['ra','dec'],tab2.colnames)), "Table 2 doesn't have column 'ra' and/or 'dec'."
+
+
+    matched_tab1, matched_tab2 = xmatch_catalogs(tab1, tab2, tol, **kwargs)
```

Comment:
Add a few more inline comments here and there.

Reply:
Done.